web/src/components/horizontalcolorbar.js

```diff
@@ -1,217 +1,83 @@
-      .attr('y', 0)
-      .attr('height', this.colorbarHeight)
-      .style('fill', (d, i) => that.colors[i]);
-  }
-
-  // Prepare an invisible marker
-  if (this.scale.ticks) {
-    this.gColorbar.select('line')
-      .attr('y2', this.colorbarHeight);
-  } else {
-    this.gColorbar.select('rect')
-      .attr('width', this.deltaOrdinal);
-  }
-
-  // Draw a container around the colorbar
-  this.gColorbar.select('rect.border')
-    .attr('width', this.colorbarWidth)
-    .attr('height', this.colorbarHeight);
-
-  // Draw the horizontal axis
-  const axis = d3.axisBottom(this.scale)
-    .tickSizeInner(this.colorbarHeight / 2.0)
-    .tickPadding(3).ticks(7);
-  if (this.d3TickFormat) { axis.tickFormat(this.d3TickFormat); }
-  if (this.d3TickValues) { axis.tickValues(this.d3TickValues); }
-
-  this.gColorbarAxis
-    .attr('class', 'x axis')
-    .attr('transform', `translate(0, ${this.colorbarHeight})`)
-    .call(axis);
-  this.gColorbarAxis.selectAll('.tick text');
-  this.gColorbarAxis.selectAll('.tick line')
-    .style('stroke-width', 1)
-    .attr('shape-rendering', 'crispEdges');
-  this.gColorbarAxis.select('path')
-    .style('fill', 'none')
-    .style('stroke', 'none');
-
-  return this;
-};
-
-HorizontalColorbar.prototype.currentMarker = function currentMarker(d) {
-  this.width = this.root.node().getBoundingClientRect().width;
-  this.height = this.root.node().getBoundingClientRect().height;
-  if (!this.width || !this.height) return this;
-
-  if (d !== undefined) {
-    if (this.scale.ticks) {
-      // Linear
-      this.gColorbar.select('.marker')
-        .attr('x1', this.scale(d))
-        .attr('x2', this.scale(d))
-        .style('display', 'block');
-    } else {
-      // Ordinal
-      this.gColorbar.select('.marker')
-        .attr('x', this.scale(d))
-        .attr('width', this.delta)
-        .style('display', 'block');
-    }
-  } else {
-    this.gColorbar.select('.marker')
-      .style('display', 'none');
-  }
-  return this;
+const spreadOverDomain = (scale, count) => {
+  const [x1, x2] = extent(scale.domain());
+  return range(count).map(v => x1 + (x2 - x1) * v / (count - 1));
 };
 
-HorizontalColorbar.prototype.markerColor = function markerColor(arg) {
-  this.gColorbar.select('.marker')
-    .style('stroke', arg);
-  return this;
-};
-
-HorizontalColorbar.prototype.domain = function domain(arg) {
-  if (!arg) return this._domain;
-  this._setDomainAndRange(arg, undefined);
-  return this;
-};
-
-HorizontalColorbar.prototype._setDomainAndRange = function _setDomainAndRange(d, r) {
-  const that = this;
-
-
-  this._domain = d;
-  this.colors = r || d3.range(10).map((i) => {
-    const e = d3.extent(that._domain);
-    const inter = d3.interpolate(e[0], e[1]);
-    const value = inter(i / (10 - 1));
-    return that.originalColorScale(value);
-  });
-  this.scale = d3.scaleLinear()
-    .range(this.colors)
-    .domain(this._domain);
+const HorizontalColorbar = ({
+  colorScale,
+  currentMarker,
```

Comment:
maybe rename to `currentValue` to make clear it's a float?

Reply:
Good point! I fixed it in https://github.com/tmrowco/electricitymap-contrib/pull/2299/commits/8d13810bfe0ca8808c7162cf20c3e441246dc5f2.

FYI I also uncovered a typo there which made me realize that solar layer markers are actually never being displayed - I opened https://github.com/tmrowco/electricitymap-contrib/issues/2305 to address that separately.